sky/shell/tracing_controller.cc

```diff
@@ -161,6 +178,12 @@ static void TracingController_DartStreamConsumer(
 }
 
 void TracingController::StopDartTracing() {
+  if (!dart_initialized_) {
```

Comment:
Can this just early return?  It feels strange that StopDartTracing does more than just stop the dart bits. :(  Presumably we should split this method into "FinalizeTracing" and "StopDartTracing"?